Add compatibility with Aeson 1.2

Starting from version `1.2` module `Data.Aeson` exports [`Options`](https://hackage.haskell.org/package/aeson-1.2.3.0/docs/Data-Aeson.html#t:Options) which clashes with one defined in Rollbar:

```
[1 of 2] Compiling Rollbar          ( src/Rollbar.hs, .stack-work/dist/x86_64-linux/Cabal-2.0.0.2/build/Rollbar.o )
             
/home/dbushev/projects/rollbar-haskell/src/Rollbar.hs:48:17: error:
    Ambiguous occurrence ‘Options’
    It could refer to either ‘Data.Aeson.Options’,
                             imported from ‘Data.Aeson’ at src/Rollbar.hs:7:1-17
                             (and originally defined in ‘aeson-1.2.3.0:Data.Aeson.Types.Internal’)
                          or ‘Rollbar.Options’, defined at src/Rollbar.hs:43:1
   |         
48 | emptyOptions :: Options
   |                 ^^^^^^^
```

--- a/rollbar.cabal
+++ b/rollbar.cabal
@@ -26,6 +26,7 @@ library
     , monad-control
     , resourcet
     , http-conduit
+    , lifted-base
 
   hs-source-dirs:      src
 

--- a/src/Rollbar.hs
+++ b/src/Rollbar.hs
@@ -22,6 +22,9 @@ import Data.Aeson hiding (Options)
 #else
 import Data.Aeson
 #endif
+#if MIN_VERSION_basic_prelude(0,7,0)
+import Control.Exception.Lifted (catch)
+#endif
 
 default (Text)
 

--- a/src/Rollbar.hs
+++ b/src/Rollbar.hs
@@ -1,10 +1,9 @@
-{-# Language TemplateHaskell, NoImplicitPrelude, OverloadedStrings, ExtendedDefaultRules, FlexibleContexts, ScopedTypeVariables #-}
+{-# Language CPP, TemplateHaskell, NoImplicitPrelude, OverloadedStrings, ExtendedDefaultRules, FlexibleContexts, ScopedTypeVariables #-}
 {-# OPTIONS_GHC -fno-warn-type-defaults #-}
 -- | Main entry point to the application.
 module Rollbar where
 
 import BasicPrelude
-import Data.Aeson
 import Data.Aeson.TH hiding (Options)
 import Data.Text (toLower, pack)
 import qualified Data.Vector as V
@@ -18,6 +17,11 @@ import Network.HTTP.Conduit
     , newManager
     , tlsManagerSettings
     , http )
+#if MIN_VERSION_aeson(1,2,0)
+import Data.Aeson hiding (Options)
+#else
+import Data.Aeson
+#endif
 
 default (Text)
 

